packages/shopware-6-client/__tests__/services/CustomerService/getCustomer.spec.ts

```diff
@@ -0,0 +1,37 @@
+import { getCustomer, update, config } from "@shopware-pwa/shopware-6-client";
+import { apiService } from "../../../src/apiService";
+import { getCustomerEndpoint } from "../../../src/endpoints";
+import { random } from "faker";
+jest.mock("../../../src/apiService");
+const mockedAxios = apiService as jest.Mocked<typeof apiService>;
+
+describe("CustomerService - getCustomer", () => {
+  let contextToken: string;
+  beforeEach(() => {
+    jest.resetAllMocks();
+    contextToken = random.uuid();
+    update({ contextToken });
+  });
+  afterEach(() => {
+    afterEach(() => {
+      expect(config.contextToken).toEqual(contextToken);
+    });
+  });
+
+  it("should an current customer's data - using correct token", async () => {
+    mockedAxios.get.mockResolvedValueOnce({
+      data: { data: { id: "c370eb5cd1df4d4dbcc78f055b693e79" } }
+    });
+    const result = await getCustomer();
+    expect(mockedAxios.get).toBeCalledTimes(1);
+    expect(mockedAxios.get).toBeCalledWith(getCustomerEndpoint());
+    expect(result.id).toEqual("c370eb5cd1df4d4dbcc78f055b693e79");
+  });
+
+  it("should an current customer's data - using incorrect token", async () => {
+    mockedAxios.get.mockRejectedValue(new Error());
+    expect(getCustomer()).rejects.toThrow();
+    expect(mockedAxios.get).toBeCalledTimes(1);
+    expect(mockedAxios.get).toBeCalledWith(getCustomerEndpoint());
```

Comment:
I am not sure if it's needed but when testing a behavior of some methods, that use incorrect token we may overwrite the correct token in config.contextToken with some invalid one. 

Right now we are testing response from the API, which is mocked to Error() but we are not sure about behavior of `getCustomer()` method. If that would be a black box method then we could assume there could be some validation, that will not even call `get` http method if token does not meet some format requirements.

Not a required change but from my perspective erasing the correct token would be a good step. :) Up to you to decide. :)

Reply:
You are totally right, I just left the 400 and promis rejection without any extra info to just break the logic from getCustomer() and end up with unhandled rejection. We will prepare the consistent error throwing/handling in the next few iterations once we discover all the cases :+1: